File: Algorithms/860.Lemonade-change/lemonade_change.go
func lemonadeChange(bills []int) bool {
    five_cnt, ten_cnt := 0, 0
    for _, value := range bills {
        switch value {
            case 5:
            five_cnt++
            case 10:
            five_cnt--
            ten_cnt++
            case 20:
            if ten_cnt > 0 {
                ten_cnt--
                five_cnt--
            } else {
                five_cnt -= 3
            }
        }
        if five_cnt < 0 || ten_cnt < 0 {
            return false
        }
    }
    return true
}

File: Algorithms/860.Lemonade-change/lemonade_change.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        five_cnt = 0
        ten_cnt = 0
        for ele in bills:
            if ele == 5:
                five_cnt += 1
            elif ele == 10:
                if five_cnt < 1:
                    return False
                five_cnt -= 1
                ten_cnt += 1
            else:
                if ten_cnt > 0:
                    ten_cnt -= 1
                    five_cnt -= 1
                else:
                    five_cnt -= 3
            if five_cnt < 0 or ten_cnt < 0:
                return False
        return True
